wallet-test/src/test/scala/org/bitcoins/wallet/ProcessBlockTest.scala

```diff
@@ -36,4 +42,42 @@ class ProcessBlockTest extends BitcoinSWalletTest {
       assert(utxos.head.txid == txId)
     }
   }
+
+  it must "process coinbase txs" in { param =>
+    val WalletWithBitcoindV19(wallet, bitcoind) = param
+    for {
+      startingUtxos <- wallet.listUtxos()
+      _ = assert(startingUtxos.isEmpty)
+      addr <- wallet.getNewAddress()
+      hashes <- bitcoind.generateToAddress(101, addr)
+      blocks <- FutureUtil.sequentially(hashes)(bitcoind.getBlockRaw)
+      _ <- FutureUtil.sequentially(blocks)(wallet.processBlock)
+      utxos <- wallet.listUtxos()
+      balance <- wallet.getBalance()
+    } yield {
+      assert(utxos.size == 101)
+      assert(balance == Bitcoins(50 * 48) + Bitcoins(25 * 53))
```

Comment:
So i don't really understand what these magic numbers are here other than the `50`, but I don't think this unit test is correct

>//starting balance
> ./bitcoin-cli -regtest getbalance
> 6088.99982620

> //get a new address
>./bitcoin-cli -regtest getnewaddress
bcrt1qhlsyvrn76s282dp07mshryy4sqsflt9a3cchtz

>//send money to that address
> ./bitcoin-cli -regtest generatetoaddress 1 bcrt1qhlsyvrn76s282dp07mshryy4sqsflt9a3cchtz
[
  "62d1c6c2c2332aa0f718187aa5dba4c570fe2dfa0ca621542125c64c2ee2d35a"
]

>//balance hasn't changed!!
>./bitcoin-cli -regtest getbalance
6088.99982620

I think this should be our behavior too? I believe ours is different?

Reply:
The `Bitcoins(50 * 48) + Bitcoins(25 * 53)` comes from, start at block height 101, then halving occurs at block 150, so we have 48 50 btc blocks, and then 53 25 btc blocks